src/MassTransit/Context/ILogContext.cs

```diff
@@ -0,0 +1,50 @@
+namespace MassTransit.Context
+{
+    using Microsoft.Extensions.Logging;
+
+
+    /// <summary>
+    /// Used to provide access to logging and diagnostic services
+    /// </summary>
+    public interface ILogContext
+    {
+        /// <summary>
+        /// The log context for all message movement, sent, received, etc.
+        /// </summary>
+        ILogContext Messages { get; }
+
+        /// <summary>
+        /// If enabled, returns a valid source which can be used
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>A valid source, or null</returns>
+        EnabledDiagnosticSource? IfEnabled(string name);
```

Comment:
how you stop, diagnostics source? (sorry a lot of files can't find)

Reply:
Stop it how? I'm not sure I understand what you mean.

Reply:
I've checked full source code, everything looks fine, sorry